src/modules/bot/bot.service.ts

```diff
@@ -663,7 +663,9 @@ export class BotService {
 
     return Promise.all(deletePromises)
     .then(() => {
-      return this.invalidateTransactionLayerCache();
+      return this.invalidateTransactionLayerCache().then(() => {
```

Comment:
What happens when `invalidateTransactionLayerCache` call fails?? Please write a test case for this in which this call fails.